Packs/FetchIndicatorsFromFile/Scripts/FetchIndicatorsFromFile/FetchIndicatorsFromFile.py

```diff
@@ -0,0 +1,176 @@
+    if sheet_name and sheet_name != 'None':
+        xl_sheet = xl_woorkbook.sheet_by_name(sheet_name)
+
+    else:
+        xl_sheet = xl_woorkbook.sheet_by_index(0)
+
+    for row_index in range(0, xl_sheet.nrows):
+        if row_index >= starting_row:
+            indicator = xl_sheet.cell(row_index, col_num)
+            indicator_list.append(indicator.value)
+
+    return indicator_list
+
+
+def xls_file_to_parsed_indicators(file_path, sheet_name, indicator_col, type_col, starting_row):
+    indicator_list = []
+
+    xl_woorkbook = xlrd.open_workbook(file_path)
+    if sheet_name and sheet_name != 'None':
+        xl_sheet = xl_woorkbook.sheet_by_name(sheet_name)
+
+    else:
+        xl_sheet = xl_woorkbook.sheet_by_index(0)
+
+    for row_index in range(0, xl_sheet.nrows):
+        if row_index >= starting_row:
+            indicator = xl_sheet.cell(row_index, indicator_col).value
+            indicator_type = xl_sheet.cell(row_index, type_col).value
+
+            indicator_list.append({
+                'Value': indicator,
+                'Type': indicator_type
+            })
+
+    return indicator_list
+
+
+def txt_file_to_indicator_list(file_path):
+    with open(file_path, "r") as fp:
+        file_data = fp.read()
+
+    indicator_list = re.split('\n|,|, ', file_data)
+    return indicator_list
+
+
+def detect_type(indicator):
+    """Infer the type of the indicator.
+            Args:
+                indicator(str): The indicator whose type we want to check.
+            Returns:
+                str. The type of the indicator.
+            """
+    if re.match(ipv4cidrRegex, indicator):
+        return FeedIndicatorType.CIDR
+
+    if re.match(ipv6cidrRegex, indicator):
+        return FeedIndicatorType.IPv6CIDR
+
+    if re.match(ipv4Regex, indicator):
+        return FeedIndicatorType.IP
+
+    if re.match(ipv6Regex, indicator):
+        return FeedIndicatorType.IPv6
+
+    if re.match(sha256Regex, indicator):
+        return FeedIndicatorType.SHA256
+
+    if re.match(urlRegex, indicator):
+        return FeedIndicatorType.URL
+
+    if re.match(md5Regex, indicator):
+        return FeedIndicatorType.MD5
+
+    if re.match(sha1Regex, indicator):
+        return FeedIndicatorType.SHA1
+
+    if re.match(emailRegex, indicator):
+        return FeedIndicatorType.Account
+
+    else:
+        return FeedIndicatorType.Domain
+
+
+def fetch_indicators_from_file(args):
+    file = demisto.getFilePath(args.get('entry_id'))
+    file_path = file['path']
+    file_name = file['name']
+    auto_detect = True if args.get('auto_detect') == 'True' else False
+    default_type = args.get('default_type')
+    limit = args.get("limit")
+    offset = int(str(args.get("offset"))) if args.get('offset') else 0
+    skip_parsing = False
+    sheet_name = args.get('sheet_name')
+    indicator_col_num = args.get('indicator_column_number')
+    type_col_num = args.get('indicator_type_column_number')
+    starting_row = args.get('starting_row')
```

Comment:
different logic:
starting_row is for excel files - from which row should I start reading the indicators, it is used to avoid table headers.

offset - refers to the indicator list itself - lets say you have a list of 500 and you put a limit of 100 on your output - you can get the next 100 by putting an offset of 100.

also remember this function supports text/doc files as well as excel files so starting_row does not have a meaning in text files.

Reply:
need to add this explanation so that users will be able to use it properly

Reply:
it is explained in the yml.
I'll add a comment in the code as well